### DISCUSSION TITLE
Client-credentials flow for Social-Sign in Identities

### DISCUSSION THREAD

**Discussion by valentinyanakiev** [ory/kratos#3245] [Q&A]
Hi there,

I have a self-hosted setup with Kratos and Oathkeeper. I have Microsoft as social sign-in provider as well. We have some applications that require device authentication. We are currently keeping a Kratos identity that we use password flow with for that - but that is not ideal as users end up 2 accounts for the same purpose. Ideally, client-credentials flow of the MS identity should be used. Is that possible with the current setup? I see there is client-credentials flow in the Ory hosted service but my understanding is that is for Hydra - which we don't have in our stack. Is my understanding correct?

Thanks in advance.

**Comment by kmherrmann** [comment]
Ory supports custom OIDC providers in kratos, which is used for Google/Facebook/etc. social login. This also supports MS Azure AD. Would that work for you?

**Reply by valentinyanakiev** [reply]
Hi @kmherrmann - I know it does, I am the one who contributed the LinkedIn workflow as well, so I am quite familiar. My question is not whether I can use OIDC, but whether non-interactive OIDC flows are there (client credentials flow in particular).

**Reply by kmherrmann** [reply]
Ha, sorry about that and thanks for contributing! 👍 

So you want to do a client credentials flow but look have AD verify the client id and secret as the devices are listed as identities there?
That's not supported to my knowledge.

**Reply by kmherrmann** [reply]
But yes, hydra does OAuth2 here, and manages the clients
